File: frontend/src/types/attendanceAdjustment.ts
export type AdjustmentStatus = 'PENDING' | 'APPROVED' | 'REJECTED' | 'CANCELLED';

export interface AttendanceAdjustment {
  id: number;
  attendanceId: number;
  employeeId: number;
  employeeCode: string;
  employeeName: string;
  departmentName: string | null;
  positionName: string | null;
  workDate: string;
  shiftName: string | null;
  shiftCode: string | null;
  originalClockIn: string | null;
  originalClockOut: string | null;
  adjustedClockIn: string | null;
  adjustedClockOut: string | null;
  reason: string;
  status: AdjustmentStatus;
  createdAt: string;
  reviewedAt: string | null;
  reviewedByEmployeeId: number | null;
  reviewedByEmployeeName: string | null;
}

export interface CreateAttendanceAdjustmentRequest {
  attendanceId: number;
  adjustedClockIn: string | null;
  adjustedClockOut: string | null;
  reason: string;
}

export interface ReviewAttendanceAdjustmentRequest {
  status: 'APPROVED' | 'REJECTED';
  reviewNote?: string;
}

export interface AdjustmentFilter {
  status?: string;
  dateFrom?: string;
  dateTo?: string;
  departmentId?: number;
  employeeId?: number;
  search?: string;
  page?: number;
  pageSize?: number;
}

export interface PagedAdjustmentResult {
  items: AttendanceAdjustment[];
  totalCount: number;
  page: number;
  pageSize: number;
  totalPages: number;
}

File: backend/src/Hrms.Domain/Entities/AttendanceAdjustment.cs
using System;
using Hrms.Domain.Common;

namespace Hrms.Domain.Entities;

public class AttendanceAdjustment : BaseEntity
{
    public long AttendanceId { get; set; }
    public long RequestedByEmployeeId { get; set; }
    public DateTime? OriginalClockIn { get; set; }
    public DateTime? OriginalClockOut { get; set; }
    public DateTime? AdjustedClockIn { get; set; }
    public DateTime? AdjustedClockOut { get; set; }
    public string Reason { get; set; } = string.Empty;
    public string Status { get; set; } = "PENDING";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? ReviewedAt { get; set; }
    public long? ReviewedByEmployeeId { get; set; }
    public long? ApprovalInstanceId { get; set; }

    // Navigation properties
    public virtual AttendanceDaily AttendanceDaily { get; set; } = null!;
    public virtual Employee RequestedByEmployee { get; set; } = null!;
    public virtual Employee? ReviewedByEmployee { get; set; }
}
